sympy/logic/boolalg.py

```diff
@@ -1304,6 +1304,87 @@ def append_symbol(arg, symbols):
             for c in clauses]
 
 
+def term_to_integer(term):
+	"""
+	Constructs an integer from its base 2 digits.
+
+	Examples
+	========
+
+	>>> from sympy.logic.boolalg import term_to_integer
+	>>> term_to_integer([1, 0, 1, 1])
+	11
+	>>> terms = [[0, 0, 0, 1], [0, 0, 1, 1],
+	...			[0, 1, 1, 1], [1, 0, 1, 1], [1, 1, 1, 1]]
+	>>> map(term_to_integer, terms)
+	[1, 3, 7, 11, 15]
+
+	"""
+
+	term = list(term)
+	return int(''.join(map(str, term)), 2)
+
+
+def integer_to_term(i, n_bits=0):
+	"""
+	Gives a list of the base 2 digits in the integer i.
+	If additional n_bits argument is passed,
+	integer_to_term() extends list of digits with zeros.
+
+	Examples
+	========
+
+	>>> from sympy.logic.boolalg import integer_to_term
+	>>> integer_to_term(14)
+	[1, 1, 1, 0]
+	>>> integer_to_term(7, 5)
+	[0, 0, 1, 1, 1]
+	>>> truthtable = [1, 0, 0, 1, 1, 1, 0, 1]
+	>>> minterms = []
+	>>> n_bits = len(truthtable).bit_length() - 1
+	>>> for i, value in enumerate(truthtable):
+	...			if value:
+	...				minterms.append(integer_to_term(i, n_bits))
+	>>> minterms
+	[[0, 0, 0], [0, 1, 1], [1, 0, 0], [1, 0, 1], [1, 1, 1]]
+
+	"""
+	n = i.bit_length() if i else 1
+	prefix = [0] * (n_bits - n if n_bits else 0)
+	return prefix + map(int, list('{0:b}'.format(i)))
+
+
+def truth_table(expr, variables):
+	"""
+	Gives a list of truth values of the boolean expression.
+
+	Examples
+	========
+	>>> from sympy.logic.boolalg import truth_table
+	>>> truth_table('a&b | b&c | c&a', ['a', 'b', 'c'])
+	[0, 0, 0, 1, 0, 1, 1, 1]
```

Comment:
Thanks for your feedback.

Truth table, i.e. complete list of all points in *B<sup>n</sup>* together with the value of a function *f* in each point is the most elementary way to define a Boolean function.

But there is a convenient, fast way to determine all truth value combinations for any number of variables. Called *lexicographical ordering*, it goes like this. Suppose there are *n* different variables. Then there will be *2<sup>n</sup>* different truth value combinations of these variables. We can replace each binary vector *(x<sub>0</sub>, x<sub>1</sub>, . . . , x<sub>n-1</sub>) in *B<sup>n</sup>* by its decimal representation *x = x<sub>0</sub>2<sup>n-1</sup> + x<sub>1</sub>2<sup>n-2</sup> + . . . + x<sub>n-1</sub>*.  So instead of writing *f(0, 0, . . . , 0)* we can write *f(0)*, instead of *f(0, 0, . . . , 1)* we can write *f(1)*, instead of *f(1, 1, . . . , 1)* we can write *f(2<sup>n-1</sup>)* and so on. This allows us to define a Boolean function as *f = [f(0), f(1), f(2), . . . , f(2<sup>n-1</sup>)]*. For instance, suppose we have a 3-variable Boolean function *f = [0, 0, 0, 1, 0, 1, 1, 1]*, then our truth table will be:

    >>> from sympy.logic.boolalg import truth_table
    >>> f = [0, 0, 0, 1, 0, 1, 1, 1]
    >>> table = zip(range(len(f)), f)
    >>> for t in table:
    ...    print(t)
    (0, 0)
    (1, 1)
    (2, 0)
    (3, 1)
    (4, 1)
    (5, 1)
    (6, 0)
    (7, 1)

It is equivalent to:

    >>> from sympy.logic.boolalg import integer_to_term
    >>> n = len(f)
    >>> n_bits = n.bit_length() - 1
    >>> table = zip(map(integer_to_term, range(n), [n_bits]*n), f)
    >>> for t in table:
    ...    print(t)
    ([0, 0, 0], 0)
    ([0, 0, 1], 1)
    ([0, 1, 0], 0)
    ([0, 1, 1], 1)
    ([1, 0, 0], 1)
    ([1, 0, 1], 1)
    ([1, 1, 0], 0)
    ([1, 1, 1], 1)

By using *lexicographical ordering* each truth value combination can be *efficiently encoded* as index of truth value.


Function <code>truth_table</code> do work on SymPy boolean expressions, not only strings:

    >>> from sympy.logic.boolalg import truth_table
    >>> from sympy.abc import a,b,c
    >>> expr = a ^ b >> c
    >>> expr
    Xor(a, Implies(b, c))
    >>> truth_table(expr, [a, b, c])
    [1, 1, 0, 1, 0, 0, 1, 0]

The variables must be given as the second argument in order to define which bit represents variable in truth value combinations. For example, <code>truth_table(a >> b, [b, a])</code> gives <code>[1, 0, 1, 1]</code> because variable <code>b</code> represents the first bit.


<code>integer_to_term</code> can be used to obtain the *k-th* term in *n* variables like this: <code>integer_to_term(k, n)</code>. For example, 13-th term in 5 variables:

    >>> from sympy.logic.boolalg import integer_to_term
    >>> integer_to_term(13, 5)
    [0, 1, 1, 0, 1]

Also, <code>integer_to_term</code> can be used to obtain truth values of the *k-th* function of *n* variables (There are *2<sup>2<sup>n</sup></sup>* functions of *n* variables). For example, you can easily print truth values of all 16 boolean functions of 2 variables like this:

    >>> from sympy.logic.boolalg import integer_to_term
    >>> n = 2
    >>> functions = map(integer_to_term, range(16), [4]*16)
    >>> for i in range(4):
    ...    s = ''
    ...    for j in range(16):
    ...        s = s + ' | ' + str(functions[j][i])
    ...    print(s)

    | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 1 | 1 | 1 | 1 | 1 | 1 | 1 | 1
    | 0 | 0 | 0 | 0 | 1 | 1 | 1 | 1 | 0 | 0 | 0 | 0 | 1 | 1 | 1 | 1
    | 0 | 0 | 1 | 1 | 0 | 0 | 1 | 1 | 0 | 0 | 1 | 1 | 0 | 0 | 1 | 1
    | 0 | 1 | 0 | 1 | 0 | 1 | 0 | 1 | 0 | 1 | 0 | 1 | 0 | 1 | 0 | 1